control-plane/subcommand/connect-init/command.go

```diff
@@ -123,11 +123,25 @@ func (c *Command) Run(args []string) int {
 	if c.flagACLAuthMethod != "" {
 		// loginMeta is the default metadata that we pass to the consul login API.
 		loginMeta := map[string]string{"pod": fmt.Sprintf("%s/%s", c.flagPodNamespace, c.flagPodName)}
-		_, err = common.ConsulLogin(consulClient, cfg, c.flagACLAuthMethod, "", c.flagAuthMethodNamespace, c.flagBearerTokenFile, c.flagServiceAccountName, c.flagACLTokenSink, loginMeta, c.logger)
+		loginParams := common.LoginParams{
+			AuthMethod:      c.flagACLAuthMethod,
+			Namespace:       c.flagAuthMethodNamespace,
+			BearerTokenFile: c.flagBearerTokenFile,
+			TokenSinkFile:   c.flagACLTokenSink,
+			Meta:            loginMeta,
+		}
+		token, err := common.ConsulLogin(consulClient, loginParams, c.logger)
 		if err != nil {
+			if c.flagServiceAccountName == "default" {
```

Comment:
Do we need to wait for an error to make this kind of validation? Or can we put it with the validateFlag-type guard clauses at the beginning of `Run()`?  
(If so, it would also make writing a test for this easier.)

Reply:
That's a great question. I had to dig up git blame on this to find out why. Looks like we don't want to fail early or fail in general because https://github.com/hashicorp/consul-k8s/commit/8a63540547ababee7d0ff245a9b18308fa5e1dcd

We also can't give this warning message earlier because we only want to do it if there's an error so we can help the user find if there's a misconfiguration.